internal/service/pg/resource_pg_database_test.go

```diff
@@ -0,0 +1,237 @@
+package pg_test
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	acc "github.com/aiven/terraform-provider-aiven/internal/acctest"
+
+	"github.com/aiven/aiven-go-client"
+	"github.com/aiven/terraform-provider-aiven/internal/schemautil"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func init() {
+	resource.AddTestSweepers("aiven_pg_database", &resource.Sweeper{
+		Name: "aiven_database",
+		F:    acc.SweepDatabases,
+		Dependencies: []string{
+			"aiven_connection_pool",
+		},
+	})
+}
+
+func TestAccAivenPGDatabase_basic(t *testing.T) {
+	resourceName := "aiven_pg_database.foo"
+	projectName := os.Getenv("AIVEN_PROJECT_NAME")
+	rName := acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum)
+	rName2 := acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acc.TestAccPreCheck(t) },
+		ProviderFactories: acc.TestAccProviderFactories,
+		CheckDestroy:      testAccCheckAivenPGDatabaseResourceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccPGDatabaseResource(projectName, rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAivenPGDatabaseAttributes("data.aiven_pg_database.database"),
```

Comment:
Can we also validate here the `lc_ctype` and `lc_collate` fields? These should be set based on the Config.